test_conformance/integer_ops/test_integer_dot_product.cpp

```diff
@@ -0,0 +1,395 @@
+//
+// Copyright (c) 2021 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// This is needed for std::numeric_limits<>::min() and max() to work on Windows.
```

Comment:
If it is Windows-specific, should there be a `#ifdef _WIN32`?

Reply:
Done.

Reply:
For the record, there is an alternative pattern that could be used here that appears to be used in a few other places in the CTS tests: the call to `min` can be placed in parentheses so it doesn't get parsed as a macro, which is causing the issue.

Example: instead of `std::numeric_limits<DstType>::max()`, write `(std::numeric_limits<DstType>::max)()`.

I haven't gone this route because IMHO it's a little ugly and hard to follow, but it is another option.

We could also consider moving this define to a shared header where we are actually including `windows.h`, then we wouldn't need to define it in individual test files.  That'd be a bit error-prone though, since a few tests appear to include `windows.h` directly, and tests can also include `windows.h` indirectly via the OpenCL headers.